pkg/drivers/vmnet/vmnet.go

```diff
@@ -0,0 +1,250 @@
+//go:build darwin
+
+/*
+Copyright 2025 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// package vmnet provides the helper process connecting virtual machines to the
+// vmnet network.
+package vmnet
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+
+	"github.com/docker/machine/libmachine/log"
+	"github.com/docker/machine/libmachine/state"
+	"k8s.io/minikube/pkg/minikube/process"
+)
+
+const (
+	pidfileName    = "vmnet-helper.pid"
+	logfileName    = "vmnet-helper.log"
+	executablePath = "/opt/vmnet-helper/bin/vmnet-helper"
+)
+
+// Helper manages the vmnet-helper process.
+type Helper struct {
```

Comment:
does this need to be exported?

Reply:
Yes, the vfkit driver keeps a *vmnet.Helpper in the struct. This helps to call the helper in all the cases that need to access it (start(), Stop(), Kill()).

We can change the design so to have functions accepting the helper arguments but using a Helper seems to fit better with the way drivers are implemented.